@here/harp-mapview/lib/Tile.ts

```diff
@@ -256,6 +257,46 @@ interface PolygonFadingParameters extends FadingParameters {
     lineFadeFar?: number;
 }
 
+/**
+ * Proxies a Tile with a given offset (a numeric value to represent the longitudinal shift of the
+ * tile (a multiple of 360 degrees)). The given offset should be non zero, because we don't need to
+ * proxy the Tile at offset 0 (a Tile is by definition at offset 0).
+ *
+ * The TileObject array and the TextElement array are cloned and stored in this object.
+ */
+export class TileProxy {
+    // Copies of the objects in the Tile.
+    clonedObjects: TileObject[] = [];
+
+    // Track if the Tile has been cloned, note we can't clone in the constructor because the tile
+    // may not yet have the data decoded
+    isCloned: boolean = false;
```

Comment:
> note we can't clone in the constructor because the tile
    // may not yet have the data decoded

So simple tracking doesn't work too (also looking at where 'clone' is actually called).

Note, that tile can be updated and current implementation doesn't support it at all.
You should just "listen" for original tile geometry changes and clone everything.

Reply:
One proposal to track if geometry of `proxy` is corrent, is to introduce `Tile.geometryVersion`, which increased with each update of geometry.

Proxy, then can check in `willRender` (assuming it adheres to same interface as Tile!), will check if it has geometry which is up to date and if not, clone it ...

Reply:
You mean that I could add a call back into the Tile createGeometries function?

This is probably the cleanest way to do this and avoids storing an extra variable to track the geometry version as mentioned.

Reply:
Not sure if callbacks are cleanest solution. 
You have to keep track of them in some array, allow addition, removal etc, add callbacks methods here and there.

Side note, immediate clone of geometry in callback would add performance penalty as it might be too much CPU work for one frame, so it's better to postpone it a little.

Reply:
Ok, thanks for response, if that is the easiest way to do it, it sounds good.

Reply:
@zbigg, I have added geometryVersion, please see my latest commit.